### DISCUSSION TITLE
Trying to get started with tshock - Need assistance plz

### DISCUSSION THREAD

**Discussion by JS1HUNDRED** [Pryaxis/TShock#2424] [Q&A]
Hi,

I am pleading ignorance here. Though I am capable, there is no guidance, and I am a bit lost.

I have bought a hosthavoc server, and am running Terraria. Upon starting the world, we have found all of the settings and permissions are locked to us for some ridiculous reason, and there is no configuration option. People dont even have permission to OWN A CAT in the game!!

I have found that i apparently have to install and run tshock. I did that, and nothing changed. no options showed up. Upon googling it for help, there are only results and advice for rnning the server locally with downloadable tshock.

I did find some commands to try to run, such as /setup _code_. Naturally i have no way of knowing what this code is, because why would i? No code has been supplied to me when installing tshock on the server. Now i am being faced with the following message permanently;

_"Initial setup system is disabled. This incident has been logged.
If you are locked out ofd all admin accounts, ask for help on https://tshock.co/"_

I have not been given any info on admin accounts. I have read how you get a tshock-code.txt when you install it, and it gets deleted almost immediately (no way foranyone to ever know about this, so its too latefor that). I removed tshock, shut down the server, installed tshock, checked the file manager, and this file doesnt exist.

I then downloaded config.json and edited it. There was no password set. I added one and uploaded the file, and started the server, but I still get the error above.

I need to get into the settings so we can actually be allowed to play the game we bought on the seerver we are running it on. I dont know why there is a third party software blocking all the cames permissions. I just want them unlocked. How do i get into the setup system from here, please?

**Comment by csmir** [comment]
I suggest taking a look at [this](https://github.com/Pryaxis/TShock/discussions/2325) for any help on getting admin permissions on a specific account.

**Reply by JS1HUNDRED** [reply]
It says i dont have access to the /user command at all

**Comment by Arthri** [comment]
> People dont even have permission to OWN A CAT in the game!!

I'm not sure what hosthavoc is but it sounds like you already have TShock installed by default. Vanilla Terraria will not restrict your permissions in any form.

Do you have access to the console? It would be easier to control the server that way.

Assuming that you do, install this [plugin](https://github.com/Pryaxis/Vanillafier) and run the command `/vanillafy` in-game

Otherwise, try doing what Rozen said

**Reply by JS1HUNDRED** [reply]
Should i uninstall tshock then. Im pretty sure i installed it because that was apparently the only way to get to the actual setup?

**Reply by Arthri** [reply]
Are you using SSC by chance?

If no do you mind losing all accounts and TShock data?

If not then go to `tshock/` and delete `tshock.sqlite` if the file isn't there then that means you'll need access to the MySQL server which I don't know if your host provides.

The last option is access to the console, are you able to run commands directly in the console?

**Reply by JS1HUNDRED** [reply]
Ok i have removed all tshock files and restarted the server. Now when i run /setup it says i must provide a setup code. but i dont have one, and dont know where to get one

**Reply by Arthri** [reply]
There are two options, do you have access to the console?

If so, then it is printed in the console.

Otherwise go to the TShock folder under your server folder, then find `tshock-code.txt`

**Reply by JS1HUNDRED** [reply]
I dont know what SSC is.

After removing tshock files, I logged into the game, and did /login with a password, which it accepted. Then it told me the /setup system is disabled. and locked me out again;

![Screenshot 2021-08-10 100225](https://user-images.githubusercontent.com/88691097/128789028-ce1cb0fb-7024-43f0-b74a-9946ac2aab11.png)

**Reply by JS1HUNDRED** [reply]
tshock-code.txt has never existed.

**Reply by Arthri** [reply]
How did you get an account?

**Reply by Arthri** [reply]
> /register JS100 _password_

You shouldn't do that before using `/setup`, otherwise, yes, it will be disabled

**Reply by JS1HUNDRED** [reply]
well i cant use /setup

i have never been allowed to use it. it has never let me

**Reply by Arthri** [reply]
> well i cant use /setup
> 
> i have never been allowed to use it. it has never let me

You need to delete the tshock files again to delete your account

subsequently TShock will generate a `tshock-code.txt` file that contains a code.

You **must** use that code before registering an account

**Reply by JS1HUNDRED** [reply]
i tried this, by removing tshock and reinstalling it, and checking file manager, and there was no code file.

Currently tshock isnt even installed. and i have deleted all of the files, yet it is still restricting me?

**Reply by Arthri** [reply]
> i tried this, by removing tshock and reinstalling it, and checking file manager, and there was no code file.
> 
> Currently tshock isnt even installed. and i have deleted all of the files, yet it is still restricting me?

By that you mean you deleted `TShockAPI.dll`? If you uninstalled/deleted TShock it shouldn't restrict you

**Reply by Arthri** [reply]
Also, do you have access to the console?

**Reply by JS1HUNDRED** [reply]
yes i uninstalled tshock, then deleted the 3 tshock files on the root as well as everything in the tshock folder. But it still restricts me, and the files seem to have returned in the folder. doesnt really make sense

**Reply by JS1HUNDRED** [reply]
If you mean the servers console, there is a link to Weeb console, but it doesnt have a field to type

**Reply by Arthri** [reply]
> If you mean the servers console, there is a link to Weeb console, but it doesnt have a field to type

Are you able to SSH into it?

> i tried this, by removing tshock and reinstalling it, and checking file manager, and there was no code file.

The code file will generate after you first run TShock

**Reply by JS1HUNDRED** [reply]
How do i 'run tshock'? There is simply a button to install it under Mod Manager. Thats all.

![Screenshot 2021-08-10 102622](https://user-images.githubusercontent.com/88691097/128790490-f1411419-8008-40a6-94f1-fc3eb8da2be8.png)

**Reply by Arthri** [reply]
Run the server

**Reply by JS1HUNDRED** [reply]
the server is running. Ive also just lost all my items and hearts/mana

**Reply by Arthri** [reply]
> the server is running. Ive also just lost all my items and hearts/mana

Does tshock-code.txt exist now?

**Reply by JS1HUNDRED** [reply]
no

**Reply by JS1HUNDRED** [reply]
Sorry, ive removed and installed tshock 20 times. there has never been a code file.

**Reply by Arthri** [reply]
Are you able to SSH into the console?

**Reply by JS1HUNDRED** [reply]
I havent tried that yet. 

i believe i have finally managed to remove tshock fully where /help only shows a basic list. so this is progress

**Reply by JS1HUNDRED** [reply]
With tshock seemingly completely gone, i need to test whether we are now able to do the stuff that was locked, like using a dog license. unfortunately, i have wasted all my money on them and they have despawned everytime i tried to use one and wasnt allowed. so i need to get some gold up, buy one and test it. if its allowed, ill leave it like this without a mod that basically doesnt let us play the game.

Appreciate your assistance to this point.

**Reply by Arthri** [reply]
TShock itself isn't required but I suggest you set a server password for security reasons. (And to avoid random people joining)

**Reply by JS1HUNDRED** [reply]
yep, we have a password on the server. thanks
